// qdc-mini-client/src/App.tsx
import React, { useEffect, useState, useCallback } from 'react';
import { BrowserRouter, Routes, Route } from 'react-router-dom';
import './dashboard.css';
import Sidebar from './components/Sidebar';
import Header from './components/Header';
import MetricCards from './components/MetricCards';
import RecentOrders from './components/RecentOrders';
import { OrdersTrendChart, OrdersStatusChart } from './components/Charts';
import UpcomingPickups from './components/UpcomingPickups';
import { AddOrderModal } from './components/AddOrderModal';
import { PlaceholderView } from './components/PlaceholderView';

import { OrdersPage } from './components/OrdersPage';
import { CustomersPage } from './components/CustomersPage';
import { GarmentsPage } from './components/GarmentsPage';

export interface Garment {
  id: string;
  description: string;
  status: 'received' | 'in_cleaning' | 'ready' | 'delivered';
}

export interface Order {
  id: string;
  customerName: string;
  createdAt: string;
  garments: Garment[];
  dueDate: string;
  total: number;
}

export const App: React.FC = () => {
  const [orders, setOrders] = useState<Order[]>([]);
  const [loading, setLoading] = useState<boolean>(true);
  const [error, setError] = useState<string | null>(null);
  const [isModalOpen, setIsModalOpen] = useState(false);

  const fetchOrders = useCallback(async () => {
    setLoading(true);
    setError(null);
    try {
      const res = await fetch('http://localhost:3001/api/orders');
      if (!res.ok) {
        throw new Error(`HTTP ${res.status}`);
      }
      const data = (await res.json()) as Order[];
      setOrders(data);
    } catch (e: any) {
      setError(e.message || 'Failed to load orders');
    } finally {
      setLoading(false);
    }
  }, []);

  useEffect(() => {
    fetchOrders();
  }, [fetchOrders]);

  const handleOrderAdded = () => {
    setIsModalOpen(false);
    fetchOrders();
  };

  const DashboardHome = () => {
    if (loading && orders.length === 0) {
      return <div style={{ padding: '40px', textAlign: 'center' }}>Loading dashboard data...</div>;
    }
    if (error) {
      return <div style={{ padding: '40px', textAlign: 'center', color: 'red' }}>Error: {error}</div>;
    }
    if (orders.length === 0) {
      return (
        <div style={{ padding: '60px', textAlign: 'center', backgroundColor: 'white', borderRadius: '12px', marginTop: '20px' }}>
          <h2 style={{ marginBottom: '8px' }}>No Orders Yet</h2>
          <p style={{ color: '#64748b', marginBottom: '24px' }}>Get started by creating your first order.</p>
          <button onClick={() => setIsModalOpen(true)} className="btn-primary">Create Order</button>
        </div>
      );
    }

    return (
      <>
        <MetricCards orders={orders} />
        
        <div className="dashboard-grid">
          <div className="grid-col-left">
            <RecentOrders orders={orders} />
            <OrdersTrendChart orders={orders} />
          </div>
          
          <div className="grid-col-right">
            <OrdersStatusChart orders={orders} />
            <UpcomingPickups orders={orders} />
          </div>
        </div>
      </>
    );
  };

  return (
    <BrowserRouter>
      <div className="dashboard-layout">
        <Sidebar />
        <div className="main-content">
          <Header onAddOrder={() => setIsModalOpen(true)} />
          
          <div className="dashboard-scroll">
            <Routes>
              <Route path="/" element={<DashboardHome />} />
              <Route path="/orders" element={<OrdersPage orders={orders} />} />
              <Route path="/customers" element={<CustomersPage orders={orders} />} />
              <Route path="/garments" element={<GarmentsPage orders={orders} />} />
              <Route path="/deliveries" element={<PlaceholderView title="Deliveries" />} />
              <Route path="/billing" element={<PlaceholderView title="Billing" />} />
              <Route path="/reports" element={<PlaceholderView title="Reports" />} />
              <Route path="/products" element={<PlaceholderView title="Products & Services" />} />
              <Route path="/packages" element={<PlaceholderView title="Packages" />} />
              <Route path="/store-settings" element={<PlaceholderView title="Store Settings" />} />
              <Route path="/users-roles" element={<PlaceholderView title="Users & Roles" />} />
              <Route path="/settings" element={<PlaceholderView title="Settings" />} />
              <Route path="/help" element={<PlaceholderView title="Help & Support" />} />
            </Routes>
          </div>
        </div>

        {isModalOpen && (
          <AddOrderModal 
            onClose={() => setIsModalOpen(false)} 
            onSuccess={handleOrderAdded} 
          />
        )}
      </div>
    </BrowserRouter>
  );
};
