// qdc-mini-server/src/orders/orders.service.ts
import { Injectable } from '@nestjs/common';

export type GarmentStatus = 'received' | 'in_cleaning' | 'ready' | 'delivered';

export interface Garment {
  id: string;
  description: string;
  status: GarmentStatus;
}

export interface Order {
  id: string;
  customerName: string;
  createdAt: string; // ISO string
  garments: Garment[];
  dueDate: string;
  total: number;
}

// Start with empty data so dashboard only shows changes when added by user
const ORDERS: Order[] = [];

@Injectable()
export class OrdersService {
  findAll(): Order[] {
    return ORDERS;
  }

  findOne(id: string): Order | undefined {
    return ORDERS.find((o) => o.id === id);
  }

  create(orderData: Omit<Order, 'id' | 'createdAt'>): Order {
    const newOrder: Order = {
      ...orderData,
      id: `ORD-${1000 + ORDERS.length + 1}`,
      createdAt: new Date().toISOString(),
    };
    ORDERS.unshift(newOrder); // Add to beginning of array so newest is first
    return newOrder;
  }

  getGarmentStatusSummary(): { [status: string]: number } {
    const summary: { [status: string]: number } = {};
    for (const order of ORDERS) {
      for (const garment of order.garments) {
        if (!summary[garment.status]) {
          summary[garment.status] = 0;
        }
        summary[garment.status]++;
      }
    }
    return summary;
  }
}
